src/orion/executor/ray_backend.py

```diff
@@ -0,0 +1,134 @@
+import logging
+import traceback
+
+from orion.executor.base import (
+    AsyncException,
+    AsyncResult,
+    BaseExecutor,
+    ExecutorClosed,
+    Future,
+)
+
+try:
+    import ray
+
+    HAS_RAY = True
+except ImportError:
+    HAS_RAY = False
+
+logger = logging.getLogger(__name__)
+
+
+class _Future(Future):
+    def __init__(self, future):
+        self.future = future
+        self.exception = None
+
+    def get(self, timeout=None):
+        if self.exception:
+            raise self.exception
+        try:
+            return ray.get(self.future, timeout=timeout)
+        except ray.exceptions.GetTimeoutError as e:
+            raise TimeoutError() from e
+
+    def wait(self, timeout=None):
+        try:
+            ray.get(self.future, timeout=timeout)
+        except ray.exceptions.GetTimeoutError:
+            pass
+        except Exception as e:
+            self.exception = e
+
+    def ready(self):
+        obj_ready = ray.wait([self.future])
+        return len(obj_ready[0]) == 1
+
+    def successful(self):
+        # Python 3.6 raise assertion error
+        if not self.ready():
+            raise ValueError()
+
+        return self.future.successful()
+
+
+class Ray(BaseExecutor):
+    def __init__(
+        self,
+        n_workers=-1,
+        runtime_env=None,
+        **config,
+    ):
+        super().__init__(n_workers=n_workers)
+        self.initialized = False
+        if not HAS_RAY:
+            raise ImportError("Ray must be installed to use Ray executor.")
+        self.config = config
+        if not ray.is_initialized():
+            if runtime_env is None:
+                ray.init()
```

Comment:
`config` should be passed to `ray.init()`, and runtime_env would be contained within `config`.